### PR TITLE
YGJWWN

### PR DESCRIPTION
**Pull request by GergoSzabo00**

### CODE REVIEW

File: YGJWWN/beadando/module/vm/main.tf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  # Attach service account to VM
+  service_account {
+    email  = google_service_account.account.email
+    scopes = ["monitoring"]
+  }
+
+  # Create task file upon VM startup and put done inside
+  metadata_startup_script = "echo ${var.file_content} > ~/${var.filename}"
```
Comment (s1mark) [inline YGJWWN/beadando/module/vm/main.tf:42]: nice interpolation

**Review by s1mark — CHANGES_REQUESTED** [review]
Nice and clean code.
try running terraform plan and apply.
Share the results in the comment

**Comment by GergoSzabo00** [comment]
Terraform plan output:
data.google_project.project: Reading...
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.vm.google_compute_disk.disk will be created
  + resource "google_compute_disk" "disk" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "ygjwwn"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # module.vm.google_compute_instance.vm_instance will be created
  + resource "google_compute_instance" "vm_instance" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "ygjwwn"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = (known after apply)
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # module.vm.google_service_account.account will be created
  + resource "google_service_account" "account" {
      + account_id = "ygjwwn"
      + disabled   = false
      + email      = (known after apply)
      + id         = (known after apply)
      + member     = (known after apply)
      + name       = (known after apply)
      + project    = (known after apply)
      + unique_id  = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "ekke-cloud"
  + vm_external_ip = (known after apply)
  + vm_instance_id = (known after apply)

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.

**Comment by GergoSzabo00** [comment]
<h1>Terraform apply:</h1>
<p>Note: VM name and compute disk name should not be the same because upon the creation of the vm the the boot disk will have the same name as the VM and if the name of the attached disk is also my own neptun code terraform will complain about disk already being in use by another resource.</p>

```
data.google_project.project: Reading...
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.vm.google_compute_disk.disk will be created
  + resource "google_compute_disk" "disk" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "ygjwwn-disk"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # module.vm.google_compute_instance.vm_instance will be created
  + resource "google_compute_instance" "vm_instance" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "ygjwwn"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + attached_disk {
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)
        }

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # module.vm.google_service_account.account will be created
  + resource "google_service_account" "account" {
      + account_id = "ygjwwn"
      + disabled   = false
      + email      = (known after apply)
      + id         = (known after apply)
      + member     = (known after apply)
      + name       = (known after apply)
      + project    = (known after apply)
      + unique_id  = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "ekke-cloud"
  + vm_external_ip = (known after apply)
  + vm_instance_id = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.vm.google_service_account.account: Creating...
module.vm.google_compute_disk.disk: Creating...
module.vm.google_service_account.account: Creation complete after 1s [id=projects/ekke-cloud/serviceAccounts/ygjwwn@ekke-cloud.iam.gserviceaccount.com]
module.vm.google_compute_disk.disk: Still creating... [10s elapsed]
module.vm.google_compute_disk.disk: Creation complete after 12s [id=projects/ekke-cloud/zones/europe-central2-a/disks/ygjwwn-disk]
module.vm.google_compute_instance.vm_instance: Creating...
module.vm.google_compute_instance.vm_instance: Still creating... [10s elapsed]
module.vm.google_compute_instance.vm_instance: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/instances/ygjwwn]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

project_id = "ekke-cloud"
vm_external_ip = "10.186.0.49"
vm_instance_id = "projects/ekke-cloud/zones/europe-central2-a/instances/ygjwwn"
```

**Comment by GergoSzabo00** [comment]
I ran terraform destroy after, so the gcloud will be clean.

**Review by s1mark — APPROVED** [review]

**Comment by s1mark** [comment]
Good job

**Pull request merged**